core/src/main/scala/cats/instances/queue.scala

```diff
@@ -17,6 +18,14 @@ trait QueueInstances extends cats.kernel.instances.QueueInstances {
 
       def combineK[A](x: Queue[A], y: Queue[A]): Queue[A] = x ++ y
 
+      override def combineAllOptionK[A](as: IterableOnce[Queue[A]]): Option[Queue[A]] = {
+        val iter = as.iterator
+        if (iter.isEmpty) None else Some(iter.flatMap(_.iterator).to(Queue))
```

Comment:
I think we could write a function to share like:
```scala
def appendAll[F <: Iterable[A], A](as: IterableOnce[F], bldr: Builder[F, A]): bldr.type = {
  val it = as.iterator
  while(it.hasNext) {
    bldr ++= it.next()
  }
  bldr
}
```
then we could use it for all the collection types with `appendAll(as, Queue.newBuilder[A]).result()`